### PR TITLE
Add fast sigmoid activation function

### LINKED ISSUE #20: Add activation functions to be available for use

### PR DESCRIPTION
**Pull request by JakeRoggenbuck**
fixes #20

### CODE REVIEW

File: core/src/math/activation_functions.cpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#include "jml/math/activation_functions.hpp"
+#include <cmath>
+
+#include <iostream>
+
+namespace jml {
+
+class FastSigmoid : private ActivationFunction {
```
Comment (adamhutchings) [inline core/src/math/activation_functions.cpp]: Inside the `.cpp` implementation file you only need to actually paste the bodies of the methods, not the entire class. (Doing so may lead to errors, but don't quote me on that.) See model, vector, and matrix for examples.

File: core/src/math/activation_functions.cpp [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  public:
+    FastSigmoid();
+    FastSigmoid(double x) { precomputed = (1 + std::fabs(x)); }
+
+    // 0.5 * ((x / 1 + abs(x)) + 1)
+    //
+    double f(double x) { return 0.5 * x / (1 + std::fabs(x)); }
+    // g(x) = 1 + abs(x)
+    // this gets saved to precomputed
+    double f_precomp() { return 0.5 * this->x / precomputed; }
```
Comment (adamhutchings) [inline core/src/math/activation_functions.cpp]: Genuinely wondering so please explain if I'm missing something, what's the advantage to saving a precomputed value for the last run?

**Review by adamhutchings — CHANGES_REQUESTED** [review]
I don't know if this compiles because of the duplicated class definitions in the implementation file. And even if it does, DRY.

**Review by adamhutchings — APPROVED** [review]
Can we wait to merge this until #12 is done?

**Comment by adamhutchings** [comment]
@JakeRoggenbuck @Sophon96 is the build failing on windows an actual build failure or is it an artifact of the build script being broken when the commits were pushed

**Comment by Sophon96** [comment]
> @JakeRoggenbuck @Sophon96 is the build failing on windows an actual build failure or is it an artifact of the build script being broken when the commits were pushed

It's due to the latter. Should be fixed by simply merging #16

**Comment by adamhutchings** [comment]
#16 is merged. Would pushing a trivial commit fix the issue?

**Comment by adamhutchings** [comment]
Closes #20 .

**Comment by Sophon96** [comment]
> #16 is merged. Would pushing a trivial commit fix the issue?

Works!

**Comment by adamhutchings** [comment]
@JakeRoggenbuck are we good to merge right now?

**Comment by adamhutchings** [comment]
(As opposed to us adding other activation functions in this issue)

**Comment by JakeRoggenbuck** [comment]
> @JakeRoggenbuck are we good to merge right now?

yup!

**Pull request merged**